resources/js/features/users-edit/UserEdit.form.tsx

```diff
@@ -0,0 +1,161 @@
+import {createMutation, createQuery} from "@tanstack/solid-query";
+import {FelteForm, FelteSubmit} from "components/felte-form";
+import {
+  Checkbox,
+  HideableSection,
+  MODAL_STYLE_PRESETS,
+  Modal as ModalComponent,
+  TextField,
+  getTrimInputHandler,
+} from "components/ui";
+import {QueryBarrier, useLangFunc} from "components/utils";
+import {Admin} from "data-access/memo-api/groups/Admin";
+import {AdminUserResource} from "data-access/memo-api/resources/adminUser.resource";
+import {Api} from "data-access/memo-api/types";
+import {Component, createSignal} from "solid-js";
+import {z} from "zod";
+
+export namespace UserEditForm {
+  export const getSchema = () =>
+    z.object({
+      name: z.string(),
+      email: z.string(),
+      hasEmailVerified: z.boolean(),
+      hasPassword: z.boolean(),
+      password: z.string(),
+    });
+
+  export const getInitialValues = (user: AdminUserResource): Readonly<Input> => ({
+    name: user.name,
+    email: user.email || "",
+    hasEmailVerified: user.hasEmailVerified,
+    hasPassword: user.hasPassword,
+    password: "",
+  });
+
+  export type Input = z.input<ReturnType<typeof getSchema>>;
+  export type Output = z.output<ReturnType<typeof getSchema>>;
+
+  interface FormParams {
+    userId: Api.Id;
+  }
+
+  interface Props extends FormParams {
+    onSuccess?: () => void;
+  }
+
+  export const Component: Component<Props> = (props) => {
+    const t = useLangFunc();
+    const userQuery = createQuery(() => Admin.userQueryOptions(props.userId));
+    const user = () => userQuery.data;
+    const invalidate = Admin.useInvalidator();
+    const mutation = createMutation(() => ({
+      mutationFn: Admin.updateUser,
+      onSuccess() {
+        invalidate.users();
+        props.onSuccess?.();
+      },
+    }));
+
+    return (
+      <QueryBarrier queries={[userQuery]}>
+        <FelteForm
+          id="user_edit"
+          onSubmit={(values) =>
```

Comment:
skoro używasz mutateAsync, to przekaż funkcję async do onSubmit i użyj await

Reply:
Dobra, chociaż szczerze mówiąc async i await to lukier, i funkcja zwracająca promisa jest jak dla mnie dokładnie równie dobra co async funkcja. No i w ten sposób nie mogę tego wrzucić inline bo jakieś warningi bez sensu.